File: backend/src/main/scala/slaydemo/backend/battle/objects/BattleWeaponSwitchDirection.scala
package slaydemo.backend.battle.objects

enum BattleWeaponSwitchDirection {
  case Previous
  case NoSwitch
  case Next
}

object BattleWeaponSwitchDirection {
  def fromWire(value: Int): BattleWeaponSwitchDirection =
    if value < 0 then BattleWeaponSwitchDirection.Previous
    else if value > 0 then BattleWeaponSwitchDirection.Next
    else BattleWeaponSwitchDirection.NoSwitch

  def step(value: BattleWeaponSwitchDirection): Int =
    value match {
      case BattleWeaponSwitchDirection.Previous => -1
      case BattleWeaponSwitchDirection.NoSwitch => 0
      case BattleWeaponSwitchDirection.Next     => 1
    }
}

File: frontend/src/objects/battle/microservices/combat/objects/weapon/BattleWeaponSwitchDirection.ts
export type BattleWeaponSwitchDirection = "previous" | "no_switch" | "next";
export type BattleWeaponSwitchDirectionStep = -1 | 0 | 1;

export function battleWeaponSwitchDirectionFromWire(value: number): BattleWeaponSwitchDirection {
  if (value < 0) {
    return "previous";
  }

  if (value > 0) {
    return "next";
  }

  return "no_switch";
}

export function battleWeaponSwitchDirectionStep(value: BattleWeaponSwitchDirection): BattleWeaponSwitchDirectionStep {
  switch (value) {
    case "previous":
      return -1;
    case "next":
      return 1;
    case "no_switch":
      return 0;
  }
}

